src/helpers/chatGPT.ts

```diff
@@ -1,17 +1,27 @@
 import { removeNewlinesAndExtractValues } from "./utils";
-import { PROMPT } from "./prompt";
+import { PROMPT, TRAINING } from "./prompt";
 import { ErrorType } from "../types/Basic";
 
 export const completeGPT3 = async (messageText: string) => {
   try {
-    const apiKey = process.env.OPENAI_API_KEY;
-    const apiUrl = "https://api.openai.com/v1/completions";
+    const apiKey = OPENAI_API_KEY;
+    const apiUrl = "https://api.openai.com/v1/chat/completions";
 
     const requestBody = {
-      model: "text-davinci-003",
-      prompt: PROMPT.replace(/{messageText}/g, messageText),
-      max_tokens: 1000,
-      temperature: 1, // Adjust temperature as needed
+      model: "gpt-3.5-turbo",
+      messages: [
+        {
+          role: "system",
+          content: TRAINING,
```

Comment:
why did you write `TRAINING`

Reply:
`TRAINING` is from the new chatgpt implementation.. there is a field for training and another to pass the prompt

Both of these values are defined in the prompt file

Reply:
This looks like you're abusing training. I also asked ChatGPT its opinion and it agrees.

Yes, it appears that there might be some misunderstanding or misuse of the training and prompt in the conversation.

In the context of the OpenAI GPT-3 API, there is typically no separate field for "training" as it is mentioned here. The model is already pretrained, and you generally use the prompt to guide the model's response based on the context you provide.

The `TRAINING` constant that is being used here as a system message seems to be providing context to the conversation, explaining what the model is supposed to be doing. While it's possible to include a system-level instruction in the conversation, the content of the `TRAINING` string seems more like a description of the task rather than something that guides the model's behavior. Including this as a system message might not necessarily have the desired effect.

The way the `TRAINING` and `PROMPT` are constructed seems more suited for human understanding rather than providing meaningful context to the model. You might consider combining the content of both into one prompt that clearly defines the task and context for the model.

Here's a possible restructuring:

```typescript
export const PROMPT = `
  You are a helpful bot that assists with GitHub issues. Please determine if the following user message is suitable for a GitHub issue.

  Consider the following criteria:
  // rest of the content here
  **User Message:** "{messageText}"
`;
```

And then, the `requestBody` object would contain:

```typescript
const requestBody = {
  model: "gpt-3.5-turbo",
  prompt: PROMPT.replace(/{messageText}/g, messageText),
  max_tokens: 1500,
  temperature: 0,
};
```

By merging the content of `TRAINING` into `PROMPT`, you provide all the necessary instructions in a single prompt, which is more consistent with how the GPT-3 API typically works.

Reply:
This is wrong @pavlovcik, `TRAINING` is just a way i defined the constant..

In gpt-3.5, there is a field for `user` prompt and `system` prompt.. The system prompt is used to tell the bot what to be and how to interact and the user will contain the user input

Reply:
> I think it makes more sense to rename the variables to be `PROMPT_USER` and `PROMPT_SYSTEM` because "training" implies machine learning, which we do have in our roadmap. Clear variable naming increases maintainability.

Okay, i'd rename